src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/WaasClient.scala

```diff
@@ -0,0 +1,46 @@
+package org.broadinstitute.dsde.agora.server.dataaccess
+
+import cromwell.client.{ApiClient, ApiException}
+import cromwell.client.api.WomtoolApi
+import cromwell.client.model.WorkflowDescription
+import org.broadinstitute.dsde.agora.server.AgoraConfig
+import org.broadinstitute.dsde.agora.server.exceptions.ValidationException
+import scala.util.{Failure, Success, Try}
+
+
+object WaasClient {
+  private def womtoolApi(accessToken: String) = {
+    val apiClient = new ApiClient()
+    apiClient.setAccessToken(accessToken)
+    apiClient.setBasePath(AgoraConfig.waasServer.get)
+
+    // Really useful to see what HTTP requests/responses calls are being made...
+    // apiClient.setDebugging(true)
+
+    new WomtoolApi(apiClient)
+  }
+
+  def validate(payload: String, accessToken: String): Try[String] = {
+    // is there a cleaner way to do this?
+    try {
+      val wd = womtoolApi(accessToken).describe("v1", payload, null, null, null, null)
+
+      if (wd.getValidWorkflow) {
+        Success("OK")
+      } else {
+        Failure(ValidationException(String.join("\n", wd.getErrors())))
+      }
+    } catch {
+      case e: ApiException => Failure(e)
+    }
+  }
+
+  def describe(wdl: String, accessToken: String) :Try[WorkflowDescription] = {
+    Try {
+      womtoolApi(accessToken).describe("v1", wdl, null, null, null, null)
+    } recoverWith {
+      case e: ApiException => Failure(e)
+    }
+  }
```

Comment:
how about this for validate and describe?
```
  def describe(payload: String, accessToken: String) :Try[WorkflowDescription] =
    Try(womtoolApi(accessToken).describe("v1", payload, null, null, null, null))

  def validate(payload: String, accessToken: String): Try[String] = {
    describe(payload, accessToken) match {
      case Success(wd) if wd.getValidWorkflow => Success("OK")
      case Success(wd) =>
        // newline-delimited string of errors may not be user-friendly in the front end
        Failure(ValidationException(String.join("\n", wd.getErrors)))
      case Failure(ex) =>
        // do you want to log anything here? if not, could delete this case entirely
        Failure(ex)
    }
  }
```

Reply:
I definitely prefer the use of `Try()` over `try{...}`, and I'd like to see `validate` call `describe` instead of copy-pasting the `womtoolApi` invocation in both places.

Reply:
👍